courses/management/commands/manage_program_certificates.py

```diff
@@ -0,0 +1,75 @@
+"""
+Management command to sync grades and certificates for a course run
+"""
+from django.core.management.base import BaseCommand, CommandError
+from django.db.models import Q
+
+from courses.models import ProgramEnrollment
+from courses.utils import generate_program_certificate
+from users.api import fetch_user
+
+
+class Command(BaseCommand):
+    """
+    Command to create program certificates for users.
+    """
+
+    help = "Create program certifificates, specific or all, for user(s)."
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--user",
+            type=str,
+            help="The id, email or username of the enrolled User",
+            required=False,
+        )
+        parser.add_argument(
+            "--readable_id",
+            type=str,
+            help="The 'readable_id' value for a Program",
+            required=True,
+        )
+        super().add_arguments(parser)
+
+    def handle(
+        self, *args, **options
+    ):  # pylint: disable=too-many-locals,too-many-branches
+        """Handle command execution"""
+        program = options.get("readable_id")
+        if not program:
+            raise CommandError("Please provide a valid program readable_id.")
+
+        user = options.get("user") and fetch_user(options["user"])
+        base_query = (
+            Q(program__readable_id=program, user=user)
+            if user
+            else Q(program__readable_id=program)
+        )
+
+        enrollments = ProgramEnrollment.objects.filter(base_query)
+        if not enrollments:
+            raise CommandError(
+                f"Could not find program enrollment(s) with readable_id={program}"
+            )
+
+        results = []
+        for enrollment in enrollments:
+            user = enrollment.user
+            cert, is_created = generate_program_certificate(user, enrollment.program)
+
+            if not cert and not is_created:
+                self.stdout.write(
+                    self.style.ERROR(
+                        f"Certificate creation failed for {user.username} ({user.email}) in program {enrollment.program} due to incomplete courses"
```

Comment:
This can still look like a success message with a certificate="Ignored" type text, Since this is less of a failure than a warning. Take a look at course certificate creation command to have a better idea.